esp-hal/src/i2c/master/mod.rs

```diff
@@ -589,6 +564,19 @@ impl<'d> I2c<'d, Blocking> {
     /// Create a new I2C instance
     /// This will enable the peripheral but the peripheral won't get
```

Comment:
Not in this PR, but I don't think this is correct, it has a drop guard inside the driver, @bugadani maybe this is why you created that I2C issue during analysis?

Reply:
Could be, I agree that it doesn't look correct anymore.

Reply:
@playfulFence would you mind removing these lines about the peripheral not being reset on drop?